diagnostic.js

```diff
@@ -32,29 +32,82 @@ const Movement = require('./models/movement.js');
 
 /// ADD YOUR CODE BELOW
 
-const create = (name, description, startYear, endYear) => {};
-// Success -> console.log new Movement as JSON
-// Failure -> Console.error
+// I am not sure if this is correct, or if we have to add the string
+// max length, or ancestor from the Schema
+const create = (name, description, startYear, endYear) => {
+  Movement.create({
+    'name': name,
+    'description': description,
+    'startYear': startYear,
+    'endYear': endYear
+  }).then(function(movement) {
+    console.log(movement);
+  }).catch(function(error){
+    console.error(error);
+  }).then(done);
+};
 
-const index = () => {};
+// I am pretty sure following the set up of yesterdays
+// code along that this will create a search function
+// through all of the movements
+const index = () => {
+  let search = {};
+   if (arguments[0] && arguments[1]) {
+     let field = arguments[0];
+     let criterion = arguments[1];
+     if (criterion[0] === '/') {
+       let regex = new RegExp(criterion.slice(1, criterion.length - 1));
+       search[field] = regex;
+     } else {
+       search[field] = criterion;
+     }
+   }
+
+   Movement.find(search).then(function(movements){
+     movements.forEach(function(movement){
+       console.log(movement.toJSON());
+     });
+   }).catch(console.error).then(done);
+ };
 // Success -> console.log all Movements as JSON
 // Failure -> Console.error
 
-const show = (id) => {};
+const show = (id) => {
+  Movement.findById(id).then(function(movement){
+    console.log(movement.toObject());
```

Comment:
The invocation here should be `toJSON()` not `toObject`().  The existence of the document should also be checked before attempting to use it, otherwise an error will be thrown.